rero_ils/modules/loans/serializers/csv.py

```diff
@@ -0,0 +1,120 @@
+        """
+        # Transform ISO dates to human-readable strings
+        for field in ['end_date', 'request_expire_date']:
+            if field in hit:
+                hit[field] = ciso8601.parse_datetime(hit[field]).strftime("%x")
+
+        # Convert PID references to human-readable name
+        lib_loader_ref = LibrariesSearch()
+        pid_reference_fields = [
+            (lib_loader_ref, 'owning_library', 'library_pid'),
+            (lib_loader_ref, 'pickup_library', 'pickup_library_pid'),
+            (lib_loader_ref, 'transaction_library', 'transaction_library_pid'),
+            (PatronTypesSearch(), 'patron_type', 'patron_type_pid')
+        ]
+        for loader, field, pid_reference in pid_reference_fields:
+            if pid := hit.pop(pid_reference, None):
+                hit[field] = self.get_resource(loader, pid).get('name')
+
+        # document information dumping
+        if doc := self.get_resource(DocumentsSearch(), hit['document_pid']):
+            hit['document_title'] = title_format_text_head(doc.get('title'))
+
+        # Item information dumping
+        if item := self.get_resource(ItemsSearch(), hit['item_pid']['value']):
+            hit['item_call_numbers'] = '|'.join(filter(None, [
+                item.get('call_number'), item.get('second_call_number')
+            ]))
+            hit['item_barcode'] = item.get('barcode')
+
+        # Patron information's dumping
+        if patron := self.get_resource(Patron, hit['patron_pid']):
+            hit['patron_name'] = patron.formatted_name
+            hit['patron_email'] = patron.user.email
+            hit['patron_barcode'] = '|'.join(
+                patron.get('patron', {}).get('barcode', [])
+            )
+        return hit
+
+    def serialize_search(self, pid_fetcher, search_result, links=None,
+                         item_links_factory=None):
+        """Serialize a search result.
+
+        :param pid_fetcher: Persistent identifier fetcher.
+        :param search_result: Elasticsearch search result.
+        :param links: Dictionary of links to add to response.
+        :param item_links_factory: Factory function for record links.
+        """
+
+        def generate_csv():
+            """Generate the CSV content as a generator.
+
+            :returns: a generator of line to output into the csv file.
+            """
+            headers = dict.fromkeys(self.csv_included_fields)
+
+            # write the CSV output in memory
+            line = Line()
+            writer = DictWriter(line, quoting=QUOTE_ALL, fieldnames=headers)
+            writer.writeheader()
+            yield line.read()
+
+            for pids, records in self.get_chunks(search_result):
+                # TODO :: Load here all docs from 'record' using a single ES
+                #         query
+                for record in records:
+                    row_data = self.process_dict(self.transform_search_hit(
```

Comment:
my feeling is that's may cause bad performance issue and many Elasticsearch requests.

Reply:
I need to work on a system pre_load all resources for some types.
This should improve performance and save some ES query.
In next PR